modules/vault/main.tf

```diff
@@ -69,8 +69,9 @@ resource "aws_dynamodb_table" "vault-dynamodb-table" {
 // See https://www.terraform.io/docs/providers/aws/r/kms_key.html
 // ----------------------------------------------------------------------------
 resource "aws_kms_key" "kms_vault_unseal" {
-  description   = "KMS Key for bank vault unseal"
-  policy        = <<POLICY
+  description         = "KMS Key for bank vault unseal"
+  enable_key_rotation = var.enable_key_rotation
```

Comment:
So is there a hidden new feature ;-)? How come you enable this? Does this come out of running _tfsec_?
Reading on what this does, I am trying to understand what one gain by this automatic key rotation. 
What is your reasoning here? Also is this a property worth exposing on a module level? We had this discussion recently on some issues (might have been on the terraform-google-jx repo) on where one draws the line between exposing configuration options at the module level or basically saying that "if you need this change you need to clone the repo and adjust the code". If we were to expose all possible configuration options on module level things would get very complicated. 
If rotating this key once a year sensible and if, as the docs seem to imply, the rotation is transparent, then one could consider to enable it per default? 

I am not an expert on these matters. Just trying to understand this better.

Reply:
tfsec just points it out. This is part of the AWS well architected framework and helps to meet various compliance standards (https://www.cloudconformity.com/knowledge-base/aws/KMS/key-rotation-enabled.html#:~:text=Once%20enabled%2C%20the%20KMS%20Key,key)%20to%20perform%20the%20encryption.) 

The reason we should expose it is because this module provisions those resources, and the only way for people to change it would be this module or the aws console (which would defeat the purpose of having a terraform module in the first place)

I gave an option to enable or disable it mostly for new users experimenting with jx. Probably we should add a note for the production usage, where this needs to be enabled.

We can also just make it the default (just enable it), this would make the SREs/Security folks happy :)